core/arch/arm/plat-imx/drivers/tzc380.c

```diff
@@ -11,17 +11,24 @@
 #include <imx.h>
 #include <mm/core_memprot.h>
 #include <mm/generic_ram_layout.h>
+#include <config.h>
 
 void imx_configure_tzasc(void)
 {
 
-	int i;
+	int i = 0;
+	int end = 1;
 	vaddr_t addr[2] = {0};
 
 	addr[0] = core_mmu_get_va(TZASC_BASE, MEM_AREA_IO_SEC);
-	addr[1] = core_mmu_get_va(TZASC2_BASE, MEM_AREA_IO_SEC);
 
-	for (i = 0; i < 2; i++) {
+	if (IS_ENABLED(CFG_MX6Q) || IS_ENABLED(CFG_MX6D)
+	    || IS_ENABLED(CFG_MX6DL)) {
```

Comment:
i am sorry but i do not really see the point of this dynamic check.
there are multiple thing that i do not like with this but for me the most problematic one is that for SoC that does not have a second TZASC controller i still will have to define a  TZASC2_BASE so that it could compile.

Reply:
This is not a dynamic check any longer. `IS_ENABLED` is a macro which gets evaluated on compile time and the compiler will discard the branch if none of the conditions is met.

Reply:
Hello @Emantor , 
in your current pull request (unless i have missed something) it still is (compiler will optimized it off course if the compilation succeed).
to be sure i have done a small experiment as follow.
took your pr
git fetch origin pull/3017/head:tzc
apply a small patch
 git diff core/arch/arm/plat-imx/imx-regs.h
diff --git a/core/arch/arm/plat-imx/imx-regs.h b/core/arch/arm/plat-imx/imx-regs.h
index e1dafff8..99a6f2b7 100644
--- a/core/arch/arm/plat-imx/imx-regs.h
+++ b/core/arch/arm/plat-imx/imx-regs.h
@@ -51,7 +51,9 @@
 #define MMDC_P1_BASE                   0x021B4000
 #define MMDC_P1_SIZE                   0x4000
 #define TZASC_BASE                     0x21D0000
+#if !defined(CFG_MX6UL)
 #define TZASC2_BASE                    0x21D4000
+#endif
 #define UART2_BASE                     0x021E8000
 #define UART3_BASE                     0x021EC000
 #define UART4_BASE                     0x021F0000

compile
make CROSS_COMPILE=arm-linux-gnueabihf- PLATFORM=imx PLATFORM_FLAVOR=mx6ulevk CFG_TZC380=y

out/arm-plat-imx/core/arch/arm/plat-imx/drivers/tzc380.o
core/arch/arm/plat-imx/drivers/tzc380.c: In function ‘imx_configure_tzasc’:
core/arch/arm/plat-imx/drivers/tzc380.c:27:29: error: ‘TZASC2_BASE’ undeclared (first use in this function); did you mean ‘TZASC_BASE’?
   addr[1] = core_mmu_get_va(TZASC2_BASE, MEM_AREA_IO_SEC);
                             ^~~~~~~~~~~
                             TZASC_BASE
core/arch/arm/plat-imx/drivers/tzc380.c:27:29: note: each undeclared identifier is reported only once for each function it appears in
mk/compile.mk:147: recipe for target 'out/arm-plat-imx/core/arch/arm/plat-imx/drivers/tzc380.o' failed

Reply:
`imx8mqevk` is a better example which does not require a manipulation of the header beforehand.
IMHO solving this problem can be done by either introducing a define for `imx8mqevk` and `imx8mmevk` which evaluates to empty or restricting the compilation to `imx6` for now.
Solving this correctly would require something similar to the device trees within Linux, where each device reports its hardware instead of the hardware being compiled into OP-TEE.

Reply:
This PR has been updated and now uses an offset instead of the base address.